src/MySqlConnector/MySqlDecimal.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Globalization;
+
+namespace MySqlConnector;
+
+	public struct MySqlDecimal : IComparable, IComparable<MySqlDecimal>, IConvertible, IEquatable<MySqlDecimal>
+	{
+		private readonly string value;
+
+		internal MySqlDecimal(string val)
+		{
+			if (val.Length > 65)
+			{
+				throw new MySqlConversionException("Value is too large for Conversion");
+			}
+			value = val;
+		}
+
+		public decimal Value => Convert.ToDecimal(value, CultureInfo.InvariantCulture);
+
+		public int CompareTo(object? obj) => obj is MySqlDecimal other ?
+			((IComparable<MySqlDecimal>) this).CompareTo(other) :
```

Comment:
Cast shouldn't be necessary if `CompareTo` is public (which it is).

Reply:
Not needed as I am removing implementation of IComparable interface all together now,
Thanks for the review.